hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBuffer.java

```diff
@@ -21,12 +21,27 @@
 
 import java.nio.ByteBuffer;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 /** Buffer for a block chunk. */
 public interface ChunkBuffer {
 
   /** Similar to {@link ByteBuffer#allocate(int)}. */
   static ChunkBuffer allocate(int capacity) {
+    return allocate(capacity, null);
+  }
+
+  /**
+   * Similar to {@link ByteBuffer#allocate(int)}
+   * except that it can specify the increment.
+   *
+   * @param increment the increment size so that
+   *                  the entire buffer is allocated incrementally.
+   */
+  static ChunkBuffer allocate(int capacity, Integer increment) {
```

Comment:
Just curious: Why is increment an object? I am presuming that is for passing null, but what prevents us from passing a -1 , defined as a constant in the increment? or even a zero? That would allow us to fall back to older impl?

Reply:
Yes, when passing increment = null, it will fallback to the older implementation.  Passing -1 also work except that -1 becomes a special value.

Reply:
Passing -1/0 also has the benefit of avoiding unboxing overhead when passing increment to IncrementalChunkBuffer.

Reply:
Sure, let me change it to int.